### PR TITLE
[SPARK-30363][SQL][DOC]Add Documentation for refresh resources.

### PR DESCRIPTION
**Pull request by iRakson**
### What changes were proposed in this pull request?
Documentation added for refresh resources command in spark-sql.

### Why are the changes needed?
Previously, only refresh table command was documented. 

### Does this PR introduce any user-facing change?
Yes. Now users can access documentation for refresh resources command.

### How was this patch tested?
Manually.

### CODE REVIEW

**Comment by iRakson** [comment]
@maropu @cloud-fan @HyukjinKwon

**Comment by HyukjinKwon** [comment]
ok to test

File: docs/sql-ref-syntax-aux-resource-mgmt-refresh.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+`REFRESH` is used to invalidate and refresh all the cached data (and the associated metadata) for
+all Dataset, that contains the given data source path. Path matching is by prefix, i.e. "/" would
```
Comment (srowen) [inline docs/sql-ref-syntax-aux-resource-mgmt-refresh.md]: Dataset, that -> Datasets that

File: docs/sql-ref-syntax-aux-resource-mgmt-refresh.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+### Parameters
+<dl>
+ <dt><code><em>resource_name</em></code></dt>
+ <dd>The path of the resource that is to be refreshed.</dd>
+</dl>
+
+### Examples
+{% highlight sql %}
+REFRESH /tmp/test.jar;
```
Comment (srowen) [inline docs/sql-ref-syntax-aux-resource-mgmt-refresh.md]: Pardon, I just don't know the semantics well, but how does refreshing a JAR file relate to invalidating data associated with a Dataset?

Comment (maropu) [reply docs/sql-ref-syntax-aux-resource-mgmt-refresh.md]: `REFRESH` and `REFRESH TABLE` are mixed up... This is an explanation for `REFRESH` (`refreshByPath`). So, we need one another document for `REFRESH TABLE`.

Comment (maropu) [reply docs/sql-ref-syntax-aux-resource-mgmt-refresh.md]: Probably, we need more concrete examples, e.g. define a table, cache it, then refresh it by path...

Comment (maropu) [reply docs/sql-ref-syntax-aux-resource-mgmt-refresh.md]: Or, add comments like `REFRESH TABLE`: https://github.com/apache/spark/blob/master/docs/sql-ref-syntax-aux-refresh-table.md

**Review by srowen — COMMENTED** [review]
Looks reasonable. I guess there's not much to say here.

**Pull request closed without merging**